### PR TITLE
chore(monorepo): import bson-transpilers

### PR DESCRIPTION
**Pull request by mcasimir**
Import `bson-transpilers` from `https://github.com/mongodb-js/bson-transpilers`

### CODE REVIEW

File: packages/bson-transpilers/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  "scripts": {
+    "start": "node index.js",
+    "prepare": "npm run compile",
+    "precompile": "node download-antlr.js",
+    "compile": "npm run antlr4-js && npm run antlr4-py && npm run symbol-table",
+    "antlr4-js": "java -Xmx500M -cp './antlr-4.7.2-complete.jar:$CLASSPATH' org.antlr.v4.Tool  -Dlanguage=JavaScript -lib grammars -o lib/antlr -visitor -Xexact-output-dir grammars/ECMAScript.g4",
+    "antlr4-py": "java -Xmx500M -cp './antlr-4.7.2-complete.jar:$CLASSPATH' org.antlr.v4.Tool  -Dlanguage=JavaScript -lib grammars -o lib/antlr -visitor -Xexact-output-dir grammars/Python3.g4",
+    "symbol-table": "node compile-symbol-table.js",
+    "test": "mocha",
+    "prepublishOnly": "npm run compile",
```
Comment (mcasimir) [inline packages/bson-transpilers/package.json:22]: Do i need this if i have prepare?

Comment (gribnoysup) [reply packages/bson-transpilers/package.json:22]: No

File: packages/bson-transpilers/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "depcheck": "depcheck",
+    "precheck": "npm run depcheck",
+    "check": "mongodb-js-precommit './codegeneration/**/*{.js,.jsx}' './test/**/*.js' index.js"
```
Comment (gribnoysup) [inline packages/bson-transpilers/package.json:25]: Need to remove precommit before merging, it messes up local dev environment

File: packages/bson-transpilers/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  "main": "index.js",
+  "scripts": {
+    "start": "node index.js",
+    "prepare": "npm run compile",
+    "precompile": "node download-antlr.js",
+    "compile": "npm run antlr4-js && npm run antlr4-py && npm run symbol-table",
+    "antlr4-js": "java -Xmx500M -cp './antlr-4.7.2-complete.jar:$CLASSPATH' org.antlr.v4.Tool  -Dlanguage=JavaScript -lib grammars -o lib/antlr -visitor -Xexact-output-dir grammars/ECMAScript.g4",
+    "antlr4-py": "java -Xmx500M -cp './antlr-4.7.2-complete.jar:$CLASSPATH' org.antlr.v4.Tool  -Dlanguage=JavaScript -lib grammars -o lib/antlr -visitor -Xexact-output-dir grammars/Python3.g4",
+    "symbol-table": "node compile-symbol-table.js",
+    "test": "mocha",
```
Comment (gribnoysup) [inline packages/bson-transpilers/package.json:21]: We also need test-ci and check-ci in scripts

File: packages/bson-transpilers/package.json (hunk 2 above)
Comment (mcasimir) [reply packages/bson-transpilers/package.json:25]: This one (mongodb-js-precommit) or `pre-commit`?

File: packages/bson-transpilers/package.json (hunk 3 above)
Comment (mcasimir) [reply packages/bson-transpilers/package.json:21]: Do they need to do anything different?

Comment (mcasimir) [reply packages/bson-transpilers/package.json:21]: Made it do what they usually do

File: packages/bson-transpilers/package.json (hunk 2 above)
Comment (gribnoysup) [reply packages/bson-transpilers/package.json]: `mongodb-js-precommit` yeah (internally it has a dependency on pre-commit)

File: packages/bson-transpilers/package.json (hunk 3 above)
Comment (gribnoysup) [reply packages/bson-transpilers/package.json:21]: Yep, same command is fine, just for consistency so that we have separate script for ci stuff

**Comment by gribnoysup** [comment]
Seems like windows might need some additional configuration applied so that antlr can work there...

**Review by gribnoysup — APPROVED** [review]
Overall looks great and makes sense to keep the package here! Let's see if we can make antlr work on windows in CI, if not, let's skip it for now and revisit configuring the environments as a follow-up

**Comment by gribnoysup** [comment]
Also very sorry, but I think I caused package-lock conflicts by merging the bson update 🙈 so you might need to regenerate it again

**Pull request merged**